src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -126,6 +227,54 @@ public String getPrefixName()
         }
     }
 
+    private static void fillInBytes(
+        byte[] addressBytes,
+        int index,
+        String ipv6Group)
+    {
+        index = 2 * index + 1;
+        for (int i = ipv6Group.length(); i > 0; i-=2)
+        {
+            addressBytes[index--] = (byte) Integer.parseInt(
+                ipv6Group.substring(i - 2 > 0 ? i - 2 : 0, i), 16);
```

Comment:
Instead of trying to parse individual bytes, parse up to 4 bytes as a short, then fill in the high and low bytes from the short.

Reply:
done